Upstream: yes

From 820f07188b02d8a32ec986fd6e61542b04a1cc80 Mon Sep 17 00:00:00 2001
From: Alex Merry <alex.merry@kde.org>
Date: Sat, 29 Mar 2014 12:10:23 +0000
Subject: [PATCH 1/2] Explicitly declare exported interfaces of libraries

This is (a) generally a good idea, and (b) necessary with CMake
2.8.12.0, which defaults to PRIVATE linkage (when CMP0022 is NEW) if no
keywords are given to target_link_libraries().
---
 ksgrd/CMakeLists.txt             | 17 ++++++++++-------
 lsofui/CMakeLists.txt            |  8 +++++++-
 processcore/CMakeLists.txt       |  3 ++-
 processui/CMakeLists.txt         |  5 +++--
 processui/KTextEditVT.h          |  1 -
 processui/ProcessFilter.h        |  1 -
 processui/ProcessModel.h         |  1 -
 processui/ksysguardprocesslist.h |  4 +++-
 signalplotter/CMakeLists.txt     |  9 ++++++++-
 signalplotter/ksignalplotter.h   |  2 --
 10 files changed, 33 insertions(+), 18 deletions(-)

diff --git a/ksgrd/CMakeLists.txt b/ksgrd/CMakeLists.txt
index 6a63c3b..3daff17 100644
--- a/ksgrd/CMakeLists.txt
+++ b/ksgrd/CMakeLists.txt
@@ -18,13 +18,16 @@ add_library(KF5::SysGuard ALIAS ksgrd)
 target_include_directories(ksgrd PUBLIC "$<BUILD_INTERFACE:${libksysguard_SOURCE_DIR}>" "$<INSTALL_INTERFACE:${INCLUDE_INSTALL_DIR}/ksysguard/>")
 
 target_link_libraries(ksgrd
-    Qt5::Core
-    Qt5::Network
-    Qt5::Widgets
-    KF5::I18n
-    KF5::CoreAddons
-    KF5::ConfigWidgets
-    KF5::KDE4Support
+    PUBLIC
+        Qt5::Core
+        Qt5::Widgets # QWidget in SensorManager.h
+        Qt5::Network # QTcpSocket in SensorSocketAgent.h
+        KF5::ConfigCore # KConfigGroup in SensorManager.h
+    PRIVATE
+        KF5::I18n
+        KF5::CoreAddons
+        KF5::ConfigWidgets
+        KF5::KDE4Support
 )
 
 set_target_properties(ksgrd PROPERTIES VERSION ${LIBKSYSGUARD_VERSION_STRING} SOVERSION ${LIBKSYSGUARD_VERSION_MINOR} EXPORT_NAME SysGuard)
diff --git a/lsofui/CMakeLists.txt b/lsofui/CMakeLists.txt
index 0143f8a..d107611 100644
--- a/lsofui/CMakeLists.txt
+++ b/lsofui/CMakeLists.txt
@@ -12,7 +12,13 @@ qt5_wrap_ui( lsofui_LIB_SRCS
 add_library(lsofui ${lsofui_LIB_SRCS})
 add_library(KF5::LsofUi ALIAS lsofui)
 
-target_link_libraries(lsofui KF5::KDE4Support KF5::I18n Qt5::Widgets)
+target_link_libraries(lsofui
+    PUBLIC
+        Qt5::Widgets
+        KF5::KDE4Support
+    PRIVATE
+        KF5::I18n
+)
 set_target_properties(lsofui PROPERTIES VERSION ${LIBKSYSGUARD_VERSION_STRING} SOVERSION ${LIBKSYSGUARD_VERSION_MINOR} EXPORT_NAME LsofUi)
 install(TARGETS lsofui EXPORT libksysguardLibraryTargets ${INSTALL_TARGETS_DEFAULT_ARGS} )
 
diff --git a/processcore/CMakeLists.txt b/processcore/CMakeLists.txt
index fceccf8..f2a84ae 100644
--- a/processcore/CMakeLists.txt
+++ b/processcore/CMakeLists.txt
@@ -18,8 +18,9 @@ add_library(KF5::ProcessCore ALIAS processcore)
 
 target_link_libraries(processcore
    PUBLIC
-    KF5::I18n
+    Qt5::Core
    PRIVATE
+    KF5::I18n
     KF5::KDE4Support
     ${ZLIB_LIBRARIES}
 )
diff --git a/processui/CMakeLists.txt b/processui/CMakeLists.txt
index 675dc68..599640f 100644
--- a/processui/CMakeLists.txt
+++ b/processui/CMakeLists.txt
@@ -27,12 +27,13 @@ if(X11_XRes_FOUND)
 endif()
 
 target_link_libraries(processui
-    # FIXME: make some of these private?
     PUBLIC
+        KF5::ProcessCore
+        Qt5::Widgets
+    PRIVATE
         Qt5::Script
         Qt5::WebKitWidgets
         KF5::KDE4Support
-        processcore
 )
 target_include_directories(processui
     PUBLIC
diff --git a/processui/KTextEditVT.h b/processui/KTextEditVT.h
index 903948a..ccc78c4 100644
--- a/processui/KTextEditVT.h
+++ b/processui/KTextEditVT.h
@@ -24,7 +24,6 @@
 #define _KTextEditVT_h_
 
 #include <QTextEdit>
-#include <kdemacros.h>
 
 /*
  *    \class KTextEditVT
diff --git a/processui/ProcessFilter.h b/processui/ProcessFilter.h
index 4b23b5d..28a5abd 100644
--- a/processui/ProcessFilter.h
+++ b/processui/ProcessFilter.h
@@ -26,7 +26,6 @@
 
 #include <QSortFilterProxyModel>
 #include <QObject>
-#include <kdemacros.h>
 
 class QModelIndex;
 
diff --git a/processui/ProcessModel.h b/processui/ProcessModel.h
index f02ef53..a338536 100644
--- a/processui/ProcessModel.h
+++ b/processui/ProcessModel.h
@@ -26,7 +26,6 @@
 
 #include <QAbstractItemModel>
 
-#include <kdemacros.h>
 #include <processcore/processes.h>
 
 namespace KSysGuard {
diff --git a/processui/ksysguardprocesslist.h b/processui/ksysguardprocesslist.h
index ce469d5..c612b2d 100644
--- a/processui/ksysguardprocesslist.h
+++ b/processui/ksysguardprocesslist.h
@@ -26,7 +26,9 @@
 
 #include <QWidget>
 #include <QMetaType>
-#include <kapplication.h>
+
+#include <KConfigGroup>
+
 #include "ProcessModel.h"
 #include "ProcessFilter.h"
 #include <processcore/processes.h>
diff --git a/signalplotter/CMakeLists.txt b/signalplotter/CMakeLists.txt
index 1cfdb78..759aacf 100644
--- a/signalplotter/CMakeLists.txt
+++ b/signalplotter/CMakeLists.txt
@@ -9,7 +9,14 @@ set(ksignalplotter_LIB_SRCS
 add_library(ksignalplotter ${ksignalplotter_LIB_SRCS})
 add_library(KF5::SignalPlotter ALIAS ksignalplotter)
 
-target_link_libraries(ksignalplotter KF5::KDE4Support KF5::Plasma)
+target_link_libraries(ksignalplotter
+    PUBLIC
+        Qt5::Widgets
+        KF5::I18n # klocalizedstring.h
+    PRIVATE
+        KF5::Plasma
+        KF5::KDE4Support
+)
 
 set_target_properties(ksignalplotter
    PROPERTIES VERSION ${LIBKSYSGUARD_VERSION_STRING}
diff --git a/signalplotter/ksignalplotter.h b/signalplotter/ksignalplotter.h
index fcb2871..520dc03 100644
--- a/signalplotter/ksignalplotter.h
+++ b/signalplotter/ksignalplotter.h
@@ -28,8 +28,6 @@
 #include <QFont>
 #include <QWidget>
 #include <klocalizedstring.h>
-#include <kcomponentdata.h>
-#include <kdemacros.h>
 
 class QPaintEvent;
 class QResizeEvent;
-- 
1.9.1


From 62ac4829769048054ea60a6a2d03556ad2bcb07d Mon Sep 17 00:00:00 2001
From: Aleix Pol <aleixpol@kde.org>
Date: Mon, 31 Mar 2014 06:35:35 +0200
Subject: [PATCH 2/2] Install the Config.cmake files

It didn't look like it was needed from the API, a look in the code
shows it is.
---
 CMakeLists.txt | 4 +++-
 1 file changed, 3 insertions(+), 1 deletion(-)

diff --git a/CMakeLists.txt b/CMakeLists.txt
index 1d2feb3..921d795 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -68,7 +68,9 @@ ecm_configure_package_config_file(KF5SysGuardConfig.cmake.in
     "${CMAKE_CURRENT_BINARY_DIR}/KF5SysGuardConfig.cmake"
     INSTALL_DESTINATION ${CMAKECONFIG_INSTALL_DIR})
 
-install(FILES ${CMAKE_CURRENT_BINARY_DIR}/KF5SysGuardConfigVersion.cmake
+install(FILES
+		${CMAKE_CURRENT_BINARY_DIR}/KF5SysGuardConfig.cmake
+		${CMAKE_CURRENT_BINARY_DIR}/KF5SysGuardConfigVersion.cmake
         DESTINATION ${CMAKECONFIG_INSTALL_DIR})
 
 install(EXPORT libksysguardLibraryTargets
-- 
1.9.1

